File: src/mbs/application/common/id_provider.py
from abc import abstractmethod
from typing import Protocol

from mbs.domain.models import UserState, UserId, PermissionTextId


class IdProvider(Protocol):
    @abstractmethod
    def user_id(self) -> UserId:
        pass

    @abstractmethod
    def user_state(self) -> UserState:
        pass

    @abstractmethod
    def permissions(self) -> list[PermissionTextId]:
        pass

    @abstractmethod
    def is_auth(self) -> bool:
        pass

File: src/application/common/id_provider.rs
use crate::domain::models::{
    permissions::Permission,
    user_id::UserId,
    user_state::UserState
};


pub trait IdProvider {
    fn user_id(&self) -> &UserId;
    fn user_state(&self) -> &UserState;
    fn permissions(&self) -> &Vec<Permission>;
}
